components/cloud/aws/src/s3.rs

```diff
@@ -241,6 +246,15 @@ impl S3Storage {
         key.to_owned()
     }
 
+    fn maybe_strip_prefix_for_string(&self, key: String) -> String {
```

Comment:
`aws` and `gcs` are two different crates. Also IMO putting them in `cloud` is somehow strange. Anyway this is a tiny code snippet, duplicating it won't be pretty harmful.

Reply:
hmm, can you elaborate more why it's strange in the cloud crate? I was thinking creating a util file in the cloud crate and these sub crates can use it. Maybe I'm missing something. 
Also we have azure not implemented, and most likely we are going to dup it again for that. The major pain of having dup code is that people likely will make mistakes for forgetting about to change all the dup places and causing unexpected results, it will be a bit complex to maintain in the future.
let me know what you think